http/client/src/main/java/org/wildfly/security/http/client/utils/ElytronHttpClientRequestBuilder.java

```diff
@@ -0,0 +1,40 @@
+package org.wildfly.security.http.client.utils;
+
+import java.net.URI;
+import java.net.http.HttpHeaders;
+import java.net.http.HttpRequest;
+import java.util.Map;
+
+public class ElytronHttpClientRequestBuilder {
+    public static HttpRequest buildRequest(HttpRequest httpRequest, Map<String, String> headers){
+        String body = null;
+        if(httpRequest.bodyPublisher() != null){
```

Comment:
@keshav-725 just a minor, you can use httpRequest.bodyPublisher().isPresent() instead